codegen/project.go

```diff
@@ -336,10 +336,12 @@ func (p *Project) GetRustOutputProtobufPath() string {
 }
 
 func (p *Project) ChainEndpoint() (string, error) {
-	if ChainConfigByID[p.Network] == nil {
+	networks := GetRegistryNetworks()
+	net := networks[p.Network]
+	if net == nil {
```

Comment:
We should define a `Find(key string)` on `NetworkRegistry` type, the method would first look for a network with the same ID, return it if found.

If no network with this ID is found, take all networks IDs, sort them, iterate in order the network IDs, for the network, inspect all aliases, if an alias matches, return it.

Add unit tests for this. This makes me think, tests could be flaky actually if the network registry is updated because by default we fetch from latest, let's see for now.